ocw/lib/EC2.py

```diff
@@ -190,27 +198,16 @@ def cleanup_all(self) -> None:
         if PCWConfig.getBoolean('cleanup/vpc_cleanup', self._namespace):
             self.cleanup_vpcs()
 
-    def delete_elastic_ips(self, region):
-        self.log_info('Deleting elastic IPs in {}'.format(region))
-        response = self.ec2_client(region).describe_addresses()
-        for addr in response['Addresses']:
-            if 'AssociationId' in addr:
-                self.log_info('Disassosiate IP with AssociationId:{}'.format(addr['AssociationId']))
-                self.ec2_client(region).disassociate_address(AssociationId=addr['AssociationId'])
-            self.log_info('Release IP with AllocationId:{}'.format(addr['AllocationId']))
-            self.ec2_client(region).release_address(AllocationId=addr['AllocationId'])
-
-    def delete_vpc(self, region: str, vpc, vpc_id: str) -> None:
+    def delete_vpc(self, region: str, vpc, vpc_id: str):
         try:
             self.log_info('{} has no associated instances. Initializing cleanup of it', vpc)
-            self.delete_elastic_ips(region)
-            self.delete_internet_gw(vpc)
-            self.delete_routing_tables(vpc)
-            self.delete_vpc_endpoints(region, vpc_id)
+            self.delete_routing_tables(region, vpc_id)
             self.delete_security_groups(vpc)
-            self.delete_vpc_peering_connections(region, vpc_id)
             self.delete_network_acls(vpc)
             self.delete_vpc_subnets(vpc)
+            self.delete_internet_gw(vpc)
+            self.delete_vpc_endpoints(region, vpc_id)
+            self.delete_vpc_peering_connections(region, vpc_id)
             if self.dry_run:
```

Comment:
Why is this here?

Reply:
because all functions which modify something on the cloud is guarded by dry_run variable so you have single switcher which makes PCW 100% harmless , which is very useful when we have some bug in code and PCW attempt to delete something what it should not . Or when new code is written and I have only "prod" environment to verify it

Reply:
But why is it called only after the other functions?

Reply:
first we deleting all related resources and than we deleting vpc itself . 
if you asking why `if self.dry_run` not wrapping all function calls but only  exact call for deleting vpc - it is because each function internally has their own `if self.dry_run` checks . This way we doing everything what is possible w/o actually deleting resources , so when you have `dry_run = true ` you are closer to `dry_run = false` because you not skipping any querying calls but only deleting calls

Reply:
I got it now. Thank you